examples/tour.jl

```diff
@@ -3,192 +3,392 @@
+# - ...elide required fields, in which case the constructor will assume them to be `missing`
+# - ...contain any other fields in addition to the required fields; such additional fields are simply ignored
 #
-#     function Legolas._transform(::typeof(Legolas.Schema("my-schema", 1));
-#                                 a=missing, b=missing, c=missing, d=missing, e=missing,
-#                                 other...)
-#         a::Real = sin(a)
-#         b::String = string(a, b, c)
-#         c::Any = [a, b, c]
-#         d::Int = d
-#         e::Any = e
-#         return (; a, b, c, d, e, other...)
-#     end
+# Demonstrating a few of these properties:
+
+# Providing the additional non-required field `x` in the input, which is simply ignored:
+fields_with_x = (; fields..., x="x")
+@test NamedTuple(FooV1(fields_with_x)) == fields
+
+# Eliding the required field `c`, which is assigned `missing` in the output:
+foo = FooV1(; a=1.0, b="hi", d=[1, 2, 3])
+@test isequal(NamedTuple(foo), (a=1.0, b="hi", c=missing, d=[1, 2, 3]))
+
+# Providing the non-compliantly-typed field `d::Int`, inducing a `MethodError`:
+@test_throws MethodError FooV1(; a=1.0, b="hi", d=2)
+
+# Implicitly providing the non-compliantly-typed field `d::Missing`, inducing a `MethodError`:
+@test_throws MethodError FooV1(; a=1.0, b="hi")
+
+#####
+##### Custom Field Assignments
+#####
+
+# Schema authors may tailor the behavior of `row` for their schema by defining custom field
+# assignments in the schema's declaration. The `example.foo@1` declaration doesn't feature
+# any such assignments, so let's declare a new schema version `example.bar@1` that does:
+@schema "example.bar" Bar
+
+@version "example.bar@1" begin
+    x::Union{Int8,Missing} = ismissing(x) ? x : Int8(clamp(x, -128, 127))
+    y::String = string(y)
+    z::String = ismissing(z) ? string(y, '_', x) : z
+end
+
+# These assignment statements are inlined into `BarV1`'s inner constructor
+# definition, such that it is roughly equivalent to:
 #
-# This `Legolas._transform` method is invoked at the core of the `MyRow` constructor. As you
-# might have noticed, this method has two interesting properties we haven't yet demonstrated.
+#   function BarV1(; x=missing, y=missing)
+#       x::Union{Int8,Missing} = ismissing(x) ? x : Int8(clamp(x, -128, 127))
+#       y::String = string(y, '_', x)
+#       return new(x, y)
+#   end
 #
-# Here we demonstrate the first property - required fields have a `missing` value by default:
-@test isequal(MyRow(a=1.5, b="hello", c="goodbye", d=2), MyRow((; a=1.5, b="hello", c="goodbye", d=2, e=missing)))
-@test_throws MethodError MyRow(a=1.5, b="hello", c="goodbye") # correctly throws a `MethodError` when evaluating `d::Int = missing`
+# ...so that invocations `BarV1(; ...)` have the following behavior:
+@test NamedTuple(BarV1(; x=200, y=:hi)) == (x=127, y="hi", z="hi_127")
+@test isequal(NamedTuple(BarV1(; y=:hi)), (x=missing, y="hi", z="hi_missing"))
```

Comment:
(this might align exactly with what you meant, but just for clarity) 

> which means that you can't add a new non->:Missing column

the table itself can have a non-`>:Missing` non-required column and still be compliant, but adding a new `non->:Missing` required field to the schema would be a breaking change that would entail a new schema version 

(which is no different than the current Legolas release)